internal/driver/driver.go

```diff
@@ -479,8 +495,8 @@ func (d *Driver) RefreshExistingDevicePaths() {
 // Devices found as part of this discovery operation are written to the channel devices.
 func (d *Driver) Discover() error {
 	d.lc.Info("Discovery is triggered")
-
-	var devices []sdkModels.DiscoveredDevice
+	devices := make(map[string]sdkModels.DiscoveredDevice)
```

Comment:
I am wondering for discovery if it makes more sense to take a more calculated approach, where we use v4l or something to discover the /dev/videos on a bus/device level rather than a plain scan of all video paths?

Reply:
I wasn't sure if we wanted to look into a more complete refactor for discovery, but I will definitely look into it.